### DISCUSSION TITLE
Simpler syntax for setget

### DISCUSSION THREAD

**Discussion by miv391** [godotengine/godot-proposals#5008] [Scripting]
Problems with `setget`:
- Syntax for using just the getter must be quite cryptic for beginner programmers who haven't seen this kind of construct before. Even experienced programmers need to read the docs to figure out how to handle this case.
- As the code completion doesn't help, the programmer needs to remember the keyword. `setget`? `getset`?

I suggest that two separate keywords, `set` and `get`, would be used instead of `setget`.

Current:

```gdscript
var my_var = 5 setget my_var_set, my_var_get
var my_var = 5 setget my_var_set				# only the setter
var my_var = 5 setget ,my_var_get				# only the getter
```

Proposal:

```dart
var my_var = 5 set my_var_set get my_var_get
var my_var = 5 set my_var_set					# only the setter
var my_var = 5 get my_var_get					# only the getter
```

**Comment by Jayman2000** [comment]
[The `setget` keyword has already been removed from Godot 4.](https://godotengine.org/article/gdscript-progress-report-new-gdscript-now-merged)

**Comment by seppoday** [comment]
Set and get was remade in Godot 4 but to be honest I don't really know if it is easier to use in G4... Maybe this can be nice place to discuss possibilities for even better implementation?

**Reply by Calinou** [reply]
GDScript's syntax and feature set for 4.0 is now frozen, so now isn't the time to discuss breaking compatibility again :slightly_smiling_face:

**Reply by seppoday** [reply]
Totally agree. There is a lot of (other) things that should be prioritized now :)

**Comment by dalexeev** [comment]
Let's close this since the keyword `setget` has been replaced with properties `set` and `get`. 4.0 was released over a year ago and there seem to be no problems with the syntax, at least those that were in 3.x.
